### PR TITLE
bpo-40597: Allow email.contextmanager set_content() to set a null string.

### PR DESCRIPTION
**Pull request by msapiro**
https://bugs.python.org/issue40597

Automerge-Triggered-By: @maxking

### CODE REVIEW

File: Lib/email/contentmanager.py [hunk 1]
```diff
@@ -146,7 +146,8 @@ def embedded_body(lines): return linesep.join(lines) + linesep
     def normal_body(lines): return b'\n'.join(lines) + b'\n'
     if cte==None:
         # Use heuristics to decide on the "best" encoding.
-        if max(len(x) for x in lines) <= policy.max_line_length:
+        if (len(lines) == 0 or
+                max(len(x) for x in lines) <= policy.max_line_length):
```
Comment (remilapeyre) [inline Lib/email/contentmanager.py]: Thanks for the fix! What do you think about using the `default` argument for this:

```python
if max((len(x) for x in lines), default=0) <= policy.max_line_length:
```

?

Comment (msapiro) [reply Lib/email/contentmanager.py]: I think using default as you suggest is better. I'll push a new commit.

File: Lib/email/contentmanager.py [hunk 2]
```diff
@@ -146,7 +146,7 @@ def embedded_body(lines): return linesep.join(lines) + linesep
     def normal_body(lines): return b'\n'.join(lines) + b'\n'
     if cte==None:
         # Use heuristics to decide on the "best" encoding.
-        if max(len(x) for x in lines) <= policy.max_line_length:
+        if max((len(x) for x in lines), default = 0) <= policy.max_line_length:
```
Comment (remilapeyre) [inline Lib/email/contentmanager.py]:
```suggestion
        if max((len(x) for x in lines), default=0) <= policy.max_line_length:
```
NIT: PEP 8 advice not using space for keyword arguments: https://www.python.org/dev/peps/pep-0008/#other-recommendations

**Review by remilapeyre — APPROVED** [review]
Thanks!

**Review by maxking — APPROVED** [review]
Thanks Mark, LGTM!

**Pull request merged**